src/lib/logs/log_sfm_metric_descriptor.py

```diff
@@ -0,0 +1,148 @@
+#     Copyright 2020 Dynatrace LLC
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+
+LOG_SELF_MONITORING_METRIC_PREFIX = "custom.googleapis.com/dynatrace/logs"
+LOG_SELF_MONITORING_ALL_REQUESTS_METRIC_TYPE = LOG_SELF_MONITORING_METRIC_PREFIX + "/all_requests"
+LOG_SELF_MONITORING_CONNECTIVITY_METRIC_TYPE = LOG_SELF_MONITORING_METRIC_PREFIX + "/connectivity"
+LOG_SELF_MONITORING_TOO_OLD_RECORDS_METRIC_TYPE = LOG_SELF_MONITORING_METRIC_PREFIX + "/too_old_records"
+LOG_SELF_MONITORING_PARSING_ERRORS_METRIC_TYPE = LOG_SELF_MONITORING_METRIC_PREFIX + "/parsing_errors"
+LOG_SELF_MONITORING_TOO_LONG_CONTENT_METRIC_TYPE = LOG_SELF_MONITORING_METRIC_PREFIX + "/too_long_content_size"
+LOG_SELF_MONITORING_PROCESSING_TIME_METRIC_TYPE = LOG_SELF_MONITORING_METRIC_PREFIX + "/processing_time"
+LOG_SELF_MONITORING_SENDING_TIME_SIZE_METRIC_TYPE = LOG_SELF_MONITORING_METRIC_PREFIX + "/sending_time"
+
+DYNATRACE_TENANT_URL_LABEL_DESCRIPTOR = {
+    "key": "dynatrace_tenant_url",
+    "valueType": "STRING",
+    "description": "URL of Dynatrace tenant used for gcp log integration"
+}
+
+LOGS_SUBSCRIPTION_ID_LABEL_DESCRIPTOR = {
+    "key": "logs_subscription_id",
+    "valueType": "STRING",
+    "description": "Subscription id of log sink pubsub subscription"
+}
+
+LOG_SELF_MONITORING_ALL_REQUESTS_METRIC_DESCRIPTOR = {
+    "type": LOG_SELF_MONITORING_ALL_REQUESTS_METRIC_TYPE,
+    "valueType": "INT64",
+    "metricKind": "GAUGE",
```

Comment:
I think we may have a problem here with `gauge` metrics and the way `context.function_name` is extracted:
````python
        self.function_name = os.environ.get("FUNCTION_NAME", "Local")
````
Metrics self monitoring code was written for function deployment and never updated for Kuberenetes deployment. It was not an issue, as metrics have to run in single container, but for logs we are going to use horizontal scaling heavily. I suspect that containers don't have `FUNCTION_NAME` env variable, which would make instances overwrite gauge values of other instances. We should refine (or at least check) code in `src/lib/context.py:83` to make it work correctly.

Reply:
It seems like GCP doesn't allow to overwrite time series - one can be sent each 10 seconds and they need to be written in order. I added new metric label - container_name - it will prevent time series from rejecting. Unfortunately it's not possible to get cluster_name that's why I left function_name as is.